### PR TITLE
added more tests

### PR DESCRIPTION
**Pull request by yabani98**

### CODE REVIEW

**Comment by yabani98** [comment]
had to add a line in production.rb/development.rb/test.rb (  routes.default_url_options[:host] = 'localhost:3000'
)
and for names testing IT JUST DON'T WORK

File: config/database.yml [hunk 1]
```diff
@@ -17,8 +17,8 @@
 default: &default
   adapter: postgresql
   encoding: unicode
-  username: <%= ENV["DB_USERNAME"] || 'postgres' %>
-  password: <%= ENV["DB_PASSWORD"] %>
+  username: postgres     #<%= ENV["DB_USERNAME"] || 'postgres' %>
```
Comment (Sadat97) [inline config/database.yml]: mat3mlsh kda tani.... zbt al env 3ndk ashl

File: config/environments/development.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
+  routes.default_url_options[:host] = 'localhost:3000'
```
Comment (Sadat97) [inline config/environments/development.rb]: leeh?

File: config/environments/production.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
+  routes.default_url_options[:host] = 'localhost:3000'
```
Comment (Sadat97) [inline config/environments/production.rb]: nfs al leeh ali fo2?

File: db/migrate/20200530203816_add_course_group_id_to_students.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddCourseGroupIdToStudents < ActiveRecord::Migration[6.0]
+  def change
+    add_column :students, :course_group_id, :string
```
Comment (Sadat97) [inline db/migrate/20200530203816_add_course_group_id_to_students.rb:3]: remove al mafrod tkon has_many

File: db/migrate/20200530204036_add_lecturer_id_to_course_groups.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddLecturerIdToCourseGroups < ActiveRecord::Migration[6.0]
+  def change
+    add_column :course_groups, :lecturer_id, :string
```
Comment (Sadat97) [inline db/migrate/20200530204036_add_lecturer_id_to_course_groups.rb:3]: nfs al fkra ali fo2

File: db/migrate/20200531224745_add_course_id_to_lecturers.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddCourseIdToLecturers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :lecturers, :course_id, :string
```
Comment (Sadat97) [inline db/migrate/20200531224745_add_course_id_to_lecturers.rb:3]: nfs ali fkra ali fo2 ali fo2

File: app/models/attendance.rb [hunk 7]
```diff
@@ -11,6 +11,14 @@
 #  updated_at      :datetime         not null
 #
 class Attendance < ApplicationRecord
- belong_to :course_group
- belong_to :student
+ belongs_to :course_group
+ belongs_to :student
+
+
+ validates :course_id ,presence: true
```
[the remark this answers is not in the document]
Comment (yabani98) [reply app/models/attendance.rb]: there is no belongs to through rails

[the remark this answers is not in the document]
Comment (yabani98) [reply app/models/attendance.rb]: maybe has one :course through => :course_group

**Pull request merged**